Formula/mbt.rb

```diff
@@ -0,0 +1,33 @@
+class Mbt < Formula
+  desc "Multi-Target Application (MTA) build tool for Cloud Applications"
+  homepage "https://sap.github.io/cloud-mta-build-tool"
+  url "https://github.com/SAP/cloud-mta-build-tool/archive/v1.2.10.tar.gz"
+  sha256 "46d548342e2fe3abd51d4c5d5702602b835a2cfd53eb09b86a2abf45825823a4"
+  license "Apache-2.0"
+  head "https://github.com/SAP/cloud-mta-build-tool.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    target = Hardware::CPU.intel? ? "build-darwin" : "build-darwin-arm"
+
+    if Hardware::CPU.intel?
+      target = "build-darwin"
+      binary = "mbt"
+    else
+      target = "build-darwin-arm"
+      binary = "mbt_darwin_arm"
+    end
+
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/SAP/cloud-mta-build-tool").install buildpath.children
+    cd "src/github.com/SAP/cloud-mta-build-tool" do
+      system "make", target
+      bin.install "release/#{binary}" => "mbt"
+    end
+  end
+
+  test do
+    system "#{bin}/mbt", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Aded a check against the output of the version info, thanks.